<a href="https://colab.research.google.com/github/ysabelmata/Bootcamp-Python-/blob/main/Hito2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Realizar Autenticacion

In [2]:
from google.colab import auth
auth.authenticate_user()

Conectar a Bigquery e importar librerias


In [3]:
from google.cloud import bigquery
import pandas as pd

Define tu proyecto, dataset y tabla aquí


In [4]:
project_id = "labproyecto2-hipotesis"
dataset = "Dataset"
table = "vista_Hipotesis5"

Crea un cliente de BigQuery autenticado para el proyecto especificado.

In [5]:
client = bigquery.Client(project="labproyecto2-hipotesis")

Define una cadena de consulta SQL utilizando una cadena formateada (f-string) de Python.

In [6]:
query = f"""
SELECT *
FROM `{"labproyecto2-hipotesis"}.{"Dataset"}.{"vista_Hipotesis5"}`
"""

Ejecutar consulta y convertir resultados en un DataFrame de pandas

In [7]:
df = client.query(query).to_dataframe()

Mostrar las primeras filas del DataFrame

In [8]:
print(df.head())

     caracteristica  promedio_bajo  promedio_alto
0        cat_acoust   5.084278e+08   5.315170e+08
1           cat_bpm   5.109426e+08   5.239727e+08
2  cat_danceability   5.441017e+08   4.244953e+08
3        cat_energy   5.197791e+08   4.974633e+08
4    cat_instrument   5.254246e+08   4.805267e+08


Test Z. Cargar paquetes y datos

In [9]:
import pandas as pd
import numpy as np
from statsmodels.stats.proportion import proportions_ztest


Paso 1: Definir Hipotesis
H0: Hipotesis Nula-- Las diferencia de streams promedio en la categorias son estadisticamente iguales.
H1: Hipotesis Alternativa-- La diferencia de streams promedio en la categorias son significativamente diferentes.

Paso 2:Elegimos el nivel de confianza que nos da el alfa
NC= 95%

In [10]:
alfa= 0.05

Paso 3: Calcular el pvalor segun el estadistico de contraste (z sobre dos muestras)

In [13]:
prom_alto_dance=424495305.92
prom_bajo_dance=544101735.95
tot_streams_alto=485000000000000
tot_streams_bajo=485000000000000


Lo pasamos a Array porque asi lo pide la funcion
--Calculo para Promedio de streams por danceability

In [14]:
array_prom_dance=np.array([prom_alto_dance,prom_bajo_dance])
array_tot_streams= np.array([tot_streams_alto,tot_streams_bajo])

pvalor = proportions_ztest(count=array_prom_dance, nobs= array_tot_streams)[1]
print(f"{pvalor:.3f}")

0.000


Test de Wilcoxon
Cargar paquetes y datos

In [1]:
import pandas as pd
import numpy as np
from scipy.stats import wilcoxon

Paso 1: Definir Hipotesis H0: Hipotesis Nula-- Las diferencia de streams promedio en la categorias son estadisticamente iguales. H1: Hipotesis Alternativa-- La diferencia de streams promedio en la categorias son significativamente diferentes.


Paso 2:Elegimos el nivel de confianza que nos da el alfa NC= 95%

In [2]:
alfa= 0.05

Paso 3: Calcular el pvalor segun el estadistico de contraste (Wilcoxon)

In [6]:
# Datos de ejemplo: dos muestras emparejadas Promedio Alto todas las categorias
promedio_alto = [424495305.92,488424806.03,480526738.76,531516964.96,497463317.18,412181178.95,467693053.44]
total_streams = [100180892198,385224029050,113404310347,125438003732,117401342855,97274758232,110375560612]

# Realizar el test de Wilcoxon
stat, p = wilcoxon(promedio_alto, total_streams)

print('Estadístico de la prueba:', stat)
print('Valor p:', p)

Estadístico de la prueba: 0.0
Valor p: 0.015625


In [7]:
# Datos de ejemplo: dos muestras emparejadas Promedio Bajo todas las categorias
promedio_bajo = [544101735.95,522791902.58,525424591.67,508427849.60,519779065.53,548206444.94,529702486.77]
total_streams = [385224029050,370136667024,372000610901,359966917517,368003578393,388130163016,375029360636]

# Realizar el test de Wilcoxon
stat, p = wilcoxon(promedio_alto, total_streams)

print('Estadístico de la prueba:', stat)
print('Valor p:', p)

Estadístico de la prueba: 0.0
Valor p: 0.015625


Interpretación de los resultados:
Estadístico de la prueba de 0.0: Esto sugiere que las diferencias absolutas entre los pares de datos son muy pequeñas, lo que indica que los valores de promedio_bajo son consistentemente menores que los valores correspondientes en total_streams.

Valor p de 0.015625: Este valor es menor que 0.05, lo que indica que hay suficiente evidencia para rechazar la hipótesis nula. Esto sugiere que hay una diferencia significativa entre los valores de promedio_bajo y total_streams.

Conclusión:
Dado que el valor p es menor que 0.05, puedes concluir que hay una diferencia significativa entre el promedio bajo de todas las categorías y el total de streams. Esto significa que los dos conjuntos de datos no provienen de la misma distribución y que la diferencia observada entre ellos no es debida al azar

In [8]:
# Datos de ejemplo: dos muestras emparejadas Promedio de todas las categorias (solo para verificar nuestra conclusion)
promedios_cat = [544101735.95,522791902.58,525424591.67,508427849.60,519779065.53,548206444.94,529702486.77,424495305.92,488424806.03,480526738.76,531516964.96,497463317.18,412181178.95,467693053.44]
streams = [385224029050,370136667024,372000610901,359966917517,368003578393,388130163016,375029360636,100180892198,385224029050,113404310347,125438003732,117401342855,97274758232,110375560612]

# Realizar el test de Wilcoxon
stat, p = wilcoxon(promedios_cat, streams)

print('Estadístico de la prueba:', stat)
print('Valor p:', p)

Estadístico de la prueba: 0.0
Valor p: 0.0001220703125


Interpretación de los resultados:
Estadístico de la prueba de 0.0: Esto sugiere que las diferencias absolutas entre los pares de datos son muy pequeñas, lo que indica que los valores de promedios_cat son consistentemente menores que los valores correspondientes en streams.

Valor p de 0.0001220703125: Este valor es muy pequeño y menor que 0.05, lo que indica que hay suficiente evidencia para rechazar la hipótesis nula. Esto sugiere que hay una diferencia significativa entre los promedios de las categorías y los streams.

Conclusión:
Dado que el valor p es muy bajo, puedes concluir que hay una diferencia significativa entre los promedios de todas las categorías y los streams. Esto significa que los dos conjuntos de datos no provienen de la misma distribución y que la diferencia observada entre ellos no es debida al azar.




Test T
Cargar paquetes y datos

In [9]:
import pandas as pd
import numpy as np
from scipy.stats import ttest_rel

Paso 1: Definir Hipotesis H0: Hipotesis Nula-- Las diferencia de streams promedio en la categorias son estadisticamente iguales. H1: Hipotesis Alternativa-- La diferencia de streams promedio en la categorias son significativamente diferentes.

Paso 2:Elegimos el nivel de confianza que nos da el alfa NC= 95%

In [10]:
alfa= 0.05

Paso 3: Calcular el pvalor segun el estadistico de contraste (Test-T)

In [12]:
promedio_alto = [
    424495305.92, 488424806.03, 480526738.76, 531516964.96,
    497463317.18, 412181178.95, 467693053.44
]
total_streams = [
    100180892198, 385224029050, 113404310347, 125438003732,
    117401342855, 97274758232, 110375560612
]

# Realizar el test t para muestras emparejadas
stat, p = ttest_rel(promedio_alto, total_streams)

print('Estadístico de la prueba:', stat)
print('Valor p:', p)

Estadístico de la prueba: -3.793916139750271
Valor p: 0.009031872476726651


Interpretación:
Estadístico de la prueba t de -3.793916139750271: Este valor indica que hay una diferencia significativa entre las dos muestras. Un valor negativo sugiere que los valores de promedio_alto tienden a ser menores que los valores correspondientes en total_streams.

Valor p de 0.009031872476726651: Este valor es menor que el umbral común de significancia de 0.05, lo que indica que hay suficiente evidencia para rechazar la hipótesis nula. Esto sugiere que las diferencias observadas entre las dos muestras no son debidas al azar.

Conclusión:
Dado que el valor p es menor que 0.05, puedes concluir que hay una diferencia significativa entre el promedio alto de todas las categorías y el total de streams. Esto significa que los dos conjuntos de datos no provienen de la misma distribución y que la diferencia observada entre ellos es estadísticamente significativa.

En resumen, los resultados de la prueba t indican que las diferencias entre promedio_alto y total_streams son significativas y no pueden atribuirse al azar.








In [13]:
promedio_bajo = [
        544101735.95,522791902.58,525424591.67,508427849.60,
        519779065.53,548206444.94,529702486.77
]
total_streams = [
        385224029050,370136667024,372000610901,359966917517,
        368003578393,388130163016,375029360636
]
# Realizar el test t para muestras emparejadas
stat, p = ttest_rel(promedio_alto, total_streams)

print('Estadístico de la prueba:', stat)
print('Valor p:', p)

Estadístico de la prueba: -100.242496270999
Valor p: 6.642201159586205e-11


Estadístico de la prueba t: -100.242496270999

Este valor extremadamente alto en magnitud y negativo indica que hay una diferencia muy significativa entre las dos muestras. Un valor negativo sugiere que los valores de la primera muestra (en este caso, promedio_alto) tienden a ser mucho menores que los valores correspondientes en la segunda muestra (en este caso, total_streams).
Valor p: 6.642201159586205e-11

Este valor p es extremadamente pequeño, mucho menor que cualquier umbral de significancia comúnmente usado (por ejemplo, 0.05, 0.01 o incluso 0.001). Un valor p tan pequeño indica que la probabilidad de obtener una diferencia tan extrema entre las muestras por azar es casi inexistente.
Conclusión:
Dado que el valor p es mucho menor que 0.05, puedes concluir que hay una diferencia extremadamente significativa entre las dos muestras. Esto significa que los dos conjuntos de datos (promedio_alto y total_streams) no provienen de la misma distribución y que la diferencia observada entre ellos no es debida al azar.

Interpretación detallada:
La magnitud del estadístico de la prueba t (en este caso, -100.242496270999) refleja la magnitud de las diferencias observadas entre las dos muestras. Un valor tan alto sugiere una diferencia muy marcada.
El signo negativo del estadístico indica que los valores de la primera muestra (promedio_alto) son consistentemente menores que los valores correspondientes en la segunda muestra (total_streams).
El valor p extremadamente bajo refuerza la conclusión de que la diferencia observada es altamente significativa y no puede atribuirse al azar.
En resumen, estos resultados indican una diferencia altamente significativa y consistente entre los dos conjuntos de datos, con promedio_alto siendo considerablemente menor que total_streams.

In [11]:
promedios_cat = [
    544101735.95, 522791902.58, 525424591.67, 508427849.60, 519779065.53,
    548206444.94, 529702486.77, 424495305.92, 488424806.03, 480526738.76,
    531516964.96, 497463317.18, 412181178.95, 467693053.44
]
streams = [
    385224029050, 370136667024, 372000610901, 359966917517, 368003578393,
    388130163016, 375029360636, 100180892198, 385224029050, 113404310347,
    125438003732, 117401342855, 97274758232, 110375560612
]

# Realizar el test t para muestras emparejadas
stat, p = ttest_rel(promedios_cat, streams)

print('Estadístico de la prueba:', stat)
print('Valor p:', p)

Estadístico de la prueba: -7.177967943583055
Valor p: 7.176319351589879e-06


Interpretación de los resultados:
Estadístico de la prueba t: Indica la magnitud de las diferencias observadas entre las dos muestras. Un valor más alto indica diferencias más significativas.
Valor p: Indica la probabilidad de observar las diferencias obtenidas entre las dos muestras si la hipótesis nula fuera verdadera. Un valor p bajo (por ejemplo, menor que 0.05) sugiere que las diferencias observadas no son debidas al azar.
Voy a ejecutar el código para obtener el estadístico de la prueba y el valor p.

Los resultados de la prueba t para muestras emparejadas son los siguientes:

Estadístico de la prueba t: -7.177967943583055
Valor p: 7.176319351589879e-06

Interpretación de los resultados:
Estadístico de la prueba t de -7.177967943583055: Este valor indica que hay una diferencia significativa entre las dos muestras. Un valor negativo sugiere que los valores de promedios_cat tienden a ser menores que los valores correspondientes en streams.

Valor p de 7.176319351589879e-06: Este valor es extremadamente pequeño y mucho menor que 0.05, lo que indica que hay suficiente evidencia para rechazar la hipótesis nula. Esto sugiere que las diferencias observadas entre las dos muestras no son debidas al azar.

Conclusión:
Dado que el valor p es extremadamente bajo, puedes concluir que hay una diferencia significativa entre los promedios de todas las categorías y los streams. Esto significa que los dos conjuntos de datos no provienen de la misma distribución y que la diferencia observada entre ellos es estadísticamente significativa.